cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/NetworkUtils.java

```diff
@@ -289,4 +299,65 @@ private void setSecondaryPrivateIpAddressCountQueryParams(Map<String, String> qu
     }
+
+    private void setFilterValues(Map<String, String> queryParamsMap, String filterName, String filterValues, int index) {
+        String[] valuesArray = getStringsArray(filterValues, EMPTY, PIPE_DELIMITER);
+        if (isNotEmpty(valuesArray)) {
+            for (int counter = START_INDEX; counter < valuesArray.length; counter++) {
+                if (!NOT_RELEVANT.equalsIgnoreCase(getFilterValue(filterName, valuesArray[counter]))
+                        || !"-1".equals(getFilterValue(filterName, valuesArray[counter]))) {
+                    queryParamsMap.put(getFilterValueKey(index, counter), getFilterValue(filterName, valuesArray[counter].toLowerCase()));
+                    if (!NOT_RELEVANT.equalsIgnoreCase(getFilterValue(filterName, valuesArray[counter])) ||
+                            !"-1".equals(getFilterValue(filterName, valuesArray[counter]))) {
+                        queryParamsMap.put(getFilterValueKey(index, counter),
+                                getFilterValue(filterName, valuesArray[counter].toLowerCase()));
+                    }
+                }
+            }
+        }
+    }
+
+    private String getFilterValue(String filterName, String filterValue) {
+        switch (filterName) {
+            case ATTACHMENT_STATUS:
+                return AttachmentStatus.getValue(filterValue);
+            default:
+                return filterValue;
+        }
+    }
+    
+    private String getFilterNameKey(int index) {
+        return FILTER + DOT + valueOf(index + ONE) + DOT + NAME;
+    }
+
+    private String getFilterValueKey(int index, int counter) {
+        return FILTER + DOT + valueOf(index + ONE) + DOT + VALUE + DOT + valueOf(counter + ONE);
+    }
+
+    private void setDescribeNetworkInterfacesFilters(Map<String, String> queryParamsMap, InputsWrapper wrapper) {
+        String[] filterNamesArray = getArrayWithoutDuplicateEntries(wrapper.getInstanceInputs().getFilterNamesString(),
+                FILTER_NAMES_STRING, wrapper.getCommonInputs().getDelimiter());
+        String[] filterValuesArray = getStringsArray(wrapper.getInstanceInputs().getFilterValuesString(), EMPTY,
+                wrapper.getCommonInputs().getDelimiter());
+        validateAgainstDifferentArraysLength(filterNamesArray, filterValuesArray, FILTER_NAMES_STRING, FILTER_VALUES_STRING);
+        if (isNotEmpty(filterNamesArray) && isNotEmpty(filterValuesArray)) {
+            for (int index = START_INDEX; index < filterNamesArray.length; index++) {
+                String filterName = NetworkFilter.getNetworkFilter(filterNamesArray[index]);
+                queryParamsMap.put(getFilterNameKey(index), filterName);
+                setFilterValues(queryParamsMap, filterName, filterValuesArray[index], index);
+            }
+        }
+    }
+
+    public Map<String, String> getDescribeNetworkInterfacesQueryParamsMap(InputsWrapper wrapper) throws Exception {
+        Map<String, String> queryParamsMap = new LinkedHashMap<>();
```

Comment:
@victorursan In some calls (especially in Describe... when filters can be set) is necessary to preserve order on them...